promql_http_api/query.py

```diff
@@ -165,11 +165,11 @@ def make_url(self):
             url += '&time=' + time_str
         return url
 
-    def to_dataframe(self, schema: Optional[dict] = None):
+    def to_dataframe(self, schema: Optional[dict] = None, authorization_token: str = ""):
```

Comment:
This is the wrong place to insert the authentication. It should propagate together with the URL from PromqlHttpApi (api.py).

There are several reasons for that:
1. The authorization doesn't change from query to query, so it is an unnecessary burden to provide it with each call
2. Authorization is needed with any API call, so we should couple it with the URL and solve it once for all of them, instead of separately for each API

One option to fix that would be to change PromqlHttpApi constructor:
```python
    def __init__(self, url: str, authorization_header: Optional[str] = None):
        self.url = url
        self.authorization_header = authorization_header
```

It would then need to be propagated to all the APIs like so:
```python
    def query(self, *args, **kwargs) -> Query:
        '''
        Get a Query object
        '''
        return Query(self.url, self.authorization_header, *args, **kwargs)
```

And the necessary changes will need to be applied to the ApiEndpoint constructor to work.

Another, better and more scalable option would be to pack all HTTP args in one map, like so:
```python
    def __init__(self, url: str, authorization_header: Optional[str] = None):
        self.http_args = {
                'url': url,
                'authorization_header': authorization_header
        }

    def query(self, *args, **kwargs) -> Query:
        '''
        Get a Query object
        '''
        return Query(self.http_args, *args, **kwargs)

    # change the other methods similarly
```
The advantage of this approach is that it will be easier to introduce additional HTTP options in the future, if needed.

Again, the necessary changes will need to be applied to the ApiEndpoint constructor to work.

One final comment, I would prefer the None value over an empty string to indicate no authorization header.